src/middleware/index.ts

```diff
@@ -80,28 +85,36 @@ export function expressRateLimiter(
 
         const queryComplexity = getQueryTypeComplexity(queryAST, variables, typeWeightObject);
         try {
-            // process the request and conditinoally respond to client with status code 429 o
-            // r pass the request onto the next middleware function
+            // process the request and conditinoally respond to client with status code 429 or
+            // pass the request onto the next middleware function
             const rateLimiterResponse = await rateLimiter.processRequest(
                 ip,
                 requestTimestamp,
                 queryComplexity
             );
-            if (!rateLimiterResponse.success) {
-                // TODO: add a header 'Retry-After' with the time to wait untill next query will succeed
-                // FIXME: send information about query complexity, tokens, etc, to the client on rejected query
-                return res.status(429).json({ graphqlGate: rateLimiterResponse });
-            }
             res.locals.graphqlGate = {
                 timestamp: requestTimestamp,
                 complexity: queryComplexity,
                 tokens: rateLimiterResponse.tokens,
+                success: rateLimiterResponse.success,
+                depth: null,
             };
+            if (!rateLimiterResponse.success && !middlewareSetup.dark) {
+                // calculate the time the client should wait to send anouther query by comparing
+                // the differnce between tokens and complexity and multipying by the refill rate
+                const timeToWaitInMs =
```

Comment:
Good idea, but this calculation is only applicable to `TOKEN_BUCKET`. I think we should add this to `RateLimiterResponse` since it will be much easier for the calculation to occur when we have access to all of the information. 
- `FIXED_WINDOW` - would need start of last window
- `SLIDING_WINDOW_LOG` - probably easiest for this computation to be done while interacting with redis. 
- `SLIDING_WINDOW_COUNTER` - We should be able to solve for this with pure math but again the rate limiter could return this. 
- `LEAKY_BUCKET` - similar to what you're doing here. 

This number should be rounded up to the nearest millisecond.

Reply:
Ok, I'll probably leave this as a `fixme` and create an issue to calculate this once all the rate limit algorithms are done. Unless we want to task everyone with adding this to their current implementation?

Reply:
I'm fine with tackling it later since it will involve updating tests. Maybe just add a condition to check if `TOKEN_BUCKET` is being used.